sdk/python/kfp/_client.py

```diff
@@ -772,6 +772,38 @@ def upload_pipeline(
       IPython.display.display(IPython.display.HTML(html))
     return response
 
+  def upload_pipeline_version(
+    self,
+    pipeline_package_path,
+    pipeline_version_name: str,
+    pipeline_id: Optional[str] = None,
+    pipeline_name: Optional[str] = None
+  ):
+    """Uploads a new version of the pipeline to the Kubeflow Pipelines cluster.
+    Args:
+      pipeline_package_path: Local path to the pipeline package.
+      pipeline_version_name:  Name of the pipeline version to be shown in the UI.
+      pipeline_id: Optional. Id of the pipeline.
+      pipeline_name: Optional. Name of the pipeline.
+    Returns:
+      Server response object containing pipleine id and other information.
+    Throws:
+      Exception if pipeline id is not found.
+    """
+
+    if all([pipeline_id, pipeline_name]) or not any([pipeline_id, pipeline_name]):
```

Comment:
nit: We can do `if bool(pipeline_id) == bool(pipeline_name)`, which IMHO is slightly better.

Reply:
I would argue that the original version better communicates the 'all' or 'none' condition of this exception, making it easier to understand. However, if you prefer your approach I'm happy to change it.

Reply:
SG. Up to you